Code/Framework/AzToolsFramework/AzToolsFramework/ComponentMode/ComponentModeCollection.cpp

```diff
@@ -209,6 +215,14 @@ namespace AzToolsFramework
                 GetEntityContextId(), &EditorComponentModeNotifications::EnteredComponentMode,
                 m_activeComponentTypes);
 
+            // this call to activate the component mode editor state should eventually replace the bus call in
+            // ComponentModeCollection::BeginComponentMode() to EditorComponentModeNotifications::EnteredComponentMode
+            // such that all of the notifications for activating/deactivating the different editor modes are in a central location
+            if (m_viewportEditorModeTracker)
```

Comment:
question: Should this ever be null? Can we just get rid of the null checks? Could add an Assert if we really want to, but things weren't if this isn't around.

Reply:
we would have to go to all the tests and test fixtures where those objects are instantiated and provide a `ViewportEditorModeTracker`. i'm not saying it's nor possible, just not convinced that such an intrusive change is the way to go right now for this PR. kick that can down the road for another PR?

Reply:
update: i've removed the single arg constructor and refactored the logic to assume that the ptr is valid.